src/apps/js_generic/js_generic.cpp

```diff
@@ -1004,6 +967,47 @@ namespace ccfapp
             "Failed to register JS class definition for Body");
         }
       }
+    }
+
+    void execute_request(
+      const std::string& method,
+      const ccf::RESTVerb& verb,
+      EndpointContext& args)
+    {
+      const auto local_method = method.substr(method.find_first_not_of('/'));
+
+      const auto scripts = args.tx.ro(this->network.app_scripts);
+
+      // Try to find script for method
+      // - First try a script called "foo"
+      // - If that fails, try a script called "POST foo"
+      auto handler_script = scripts->get(local_method);
+      if (!handler_script)
+      {
+        const auto verb_prefixed =
+          fmt::format("{} {}", verb.c_str(), local_method);
+        handler_script = scripts->get(verb_prefixed);
+        if (!handler_script)
+        {
+          args.rpc_ctx->set_error(
+            HTTP_STATUS_NOT_FOUND,
+            ccf::errors::ResourceNotFound,
+            fmt::format(
+              "No handler script found for method '{}'.", verb_prefixed));
+          return;
+        }
+      }
+
+      // Ideally this should happen in the constructor but that leads
+      // to issues with quick's stack overflow checks.
+      // This is only safe as long as this instance of JSHandlers
+      // is only called from a single persistent thread.
```

Comment:
That's fine, the runtime can be created once per thread and owned by a unique pointer in thread-local storage for example.

Reply:
I made it thread-local, could you have a quick look if this is fine?  I also found a way to deal with the stack overflow check, see comments in the code.